shared/src/state-migrations/43-generalize-materials-personnel.ts

```diff
@@ -0,0 +1,302 @@
+}
+interface Personnel {
+    templateId?: UUID;
+    personnelType: PersonnelType;
+    typeName?: string;
+    typeAbbreviation?: string;
+}
+interface MaterialTemplate {
+    id?: UUID;
+    materialType?: MaterialType;
+    name?: string;
+}
+interface Material {
+    templateId?: UUID;
+    typeName?: string;
+    image: ImageProperties;
+}
+interface VehicleParameters {
+    materials: Material[];
+    personnel: Personnel[];
+    vehicle: Vehicle;
+}
+interface MapImageTemplate {
+    id: UUID;
+    image: ImageProperties;
+}
+interface MapImage {
+    id?: UUID;
+    image: ImageProperties;
+}
+
+type MaterialType = 'big' | 'standard';
+type PersonnelType = 'gf' | 'notarzt' | 'notSan' | 'rettSan' | 'san';
+
+const materialTypeNames: {
+    [key in MaterialType]: string;
+} = {
+    big: 'Erweitertes Material',
+    standard: 'Standardmaterial',
+};
+
+const materialTemplateIds: {
+    [key in MaterialType]: UUID;
+} = {
+    big: uuid(),
+    standard: uuid(),
+};
+
+const personnelTemplateIds: {
+    [key in PersonnelType]: UUID;
+} = {
+    gf: uuid(),
+    notarzt: uuid(),
+    notSan: uuid(),
+    rettSan: uuid(),
+    san: uuid(),
+};
+
+const personnelTypeAbbreviations: {
+    [Key in PersonnelType]: string;
+} = {
+    gf: 'GF',
+    notarzt: 'NA',
+    notSan: 'NFS',
+    rettSan: 'RS',
+    san: 'San',
+};
+
+const personnelTypeNames: {
+    [key in PersonnelType]: string;
+} = {
+    gf: 'Gruppenführer',
+    notarzt: 'Notarzt',
+    notSan: 'Notfallsanitäter',
+    rettSan: 'Rettungssanitäter',
+    san: 'Sanitäter',
+};
+
+function getTemplateIds(typedState: {
+    vehicleTemplates: VehicleTemplate[] | { [Key in UUID]: VehicleTemplate };
+    mapImageTemplates: MapImageTemplate[] | { [Key in UUID]: MapImageTemplate };
+}) {
+    const vehicleTemplateIds: {
+        [key in string]: UUID;
+    } = Object.fromEntries(
+        (typedState.vehicleTemplates as VehicleTemplate[]).map((template) => [
+            template.vehicleType,
+            template.id,
+        ])
+    );
+    const mapImageTemplateIds: {
+        [key in string]: UUID;
+    } = Object.fromEntries(
+        (typedState.mapImageTemplates as MapImageTemplate[]).map((template) => [
+            template.image.url,
+            template.id,
+        ])
+    );
+    return { vehicleTemplateIds, mapImageTemplateIds };
+}
+
+function migrateVehicleTemplate(vehicleTemplate: VehicleTemplate) {
+    vehicleTemplate.materialTemplateIds = vehicleTemplate.materials!.map(
+        (materialType) => materialTemplateIds[materialType]
+    );
+    delete vehicleTemplate.materials;
+    vehicleTemplate.personnelTemplateIds = vehicleTemplate.personnel!.map(
+        (personnelType) => personnelTemplateIds[personnelType]
+    );
+    delete vehicleTemplate.personnel;
+    return vehicleTemplate;
+}
+
+function migrateMapImage(
+    mapImage: MapImage,
+    mapImageTemplateIds: { [key in string]: UUID }
+) {
+    mapImage.id = mapImageTemplateIds[mapImage.image.url];
```

Comment:
I suppose this should be

```suggestion
    mapImage.templateId = mapImageTemplateIds[mapImage.image.url];
```

Afterwards, the migration test for `state-23 › combined-scenarios › everything-dropped-on-the-map-once-complete.json` should succeed. (You also have to change `id` to `templateId` in the `MapImage` type definition.)